src/payment/data/sagas.js

```diff
@@ -163,6 +163,7 @@ export function* handleSubmitPayment({ payload }) {
       }
 
       if (error.basket) {
+        yield call(handleDiscountCheck);
```

Comment:
if an error happens and the basket page updates the discount doesn't get reapplied

Reply:
I don't think this will work quite right... handleDiscountCheck is going to dispatch the basketDataReceived action, and then on line 167 below we're going to blow away that data with the data in error.basket.

I think you want to do basketDataReceived on line 167 first, _then_ call handleDiscountCheck.  i.e., reverse the order of these two lines.